src/features/Admin/Ads/AddNewAd/AddNewAd.tsx

```diff
@@ -0,0 +1,186 @@
+    };
+    const {
+        register,
+        handleSubmit,
+        setValue,
+        getValues,
+        formState: { errors },
+    } = useForm<IAds>();
+
+    // Format Data
+    // const appendToFormData = (data: IAds) => {
+    //     const formData = new FormData();
+    //     formData.append('room', data.room || ''); // Ensure data.room is not undefined
+    //     formData.append('discount',data.discount || 0); // Convert discount to string
+    //     formData.append('isActive', data.isActive); // Convert isActive to string
+    //     return formData;
+    // };
+
+    // Cancel Button
+    const goBack = () => {
+        navigate(-1);
+    };
+    const getAllAds = () => {
+        axios.get(`${adsUrl}`, {
+            headers: requestHeaders
+        })
+            .then((response) => {
+                // setAdsList(response.data.data.ads)
+            })
+            .catch((error) => {
+              
+            })
+    }
+
+
+    // Create New Room
+    const onSubmit: SubmitHandler<IAds> = async (data: IAds) => {
+        // const isActiveBoolean = active === 'yes';
+        // data.isActive = isActiveBoolean;
+        data.isActive = active === 'yes';
+        axios
+            .post(`${addAdsUrl}`, data, { headers: requestHeaders })
+            .then((response) => {
+                getAllAds();
+                navigate('/home/ads');
+                toast.success("Ads Add Successfully");
+            })
+            .catch((error) => {
+                toast.error("Error Occurred")
+            })
+            .finally(() => setIsLoading(false));
+    };
+
+    // Get All Rooms
+    const getAllRooms = () => {
+        axios
+            .get(`${roomsUrl}`, {
+                headers: requestHeaders,
+            })
+            .then((response) => {
+                setRoomsList(response.data.data.rooms);
+            })
+            .catch((error) => { });
+    };
+
+    useEffect(() => {
+        getAllRooms();
+    }, []);
+
+    return (
+        <>
+            <Container component="main" className={style.wrapper} sx={{ width: '100%', padding: 0 }}>
+                <Grid container justifyContent="center">
+                    <Grid item xs={12} sm={12} md={12} component={Paper} elevation={6} mb={8} mt={4} sx={{ padding: '2rem' }}>
+                        <Box
+                            sx={{
+                                my: 4,
+                                mx: 'auto',
+                                display: 'flex',
+                                flexDirection: 'column',
+                                width: '100%',
+                                maxWidth: '100%',
+                            }}
+                        >
+                            <form noValidate onSubmit={handleSubmit(onSubmit)}>
+                                <FormControl fullWidth sx={{ marginBottom: '1rem' }}>
+                                    <InputLabel id="demo-simple-select-label">Room Number</InputLabel>
```

Comment:
ids should be unique for each element. In this component, we have multiple elements with the same Ids.